clusterloader2/pkg/util/cluster.go

```diff
@@ -0,0 +1,157 @@
+/*
+Copyright 2018 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package util
+
+import (
+	"fmt"
+
+	"github.com/golang/glog"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	clientset "k8s.io/client-go/kubernetes"
+	nodectlr "k8s.io/kubernetes/pkg/controller/nodelifecycle"
+	"k8s.io/kubernetes/pkg/scheduler/algorithm/predicates"
+	schedulercache "k8s.io/kubernetes/pkg/scheduler/cache"
+)
+
+// GetNodesNumber returns number of nodes in the cluster.
+func GetNodesNumber(c clientset.Interface) (int, error) {
+	nodeList, err := c.CoreV1().Nodes().List(metav1.ListOptions{})
+	if err != nil {
+		return 0, err
+	}
+	numNodes := 0
+	for i := range nodeList.Items {
+		if isNodeSchedulable(&nodeList.Items[i]) && isNodeUntainted(&nodeList.Items[i]) {
+			numNodes++
+		}
+	}
+	return numNodes, err
+}
+
+// Node is schedulable if:
+// 1) doesn't have "unschedulable" field set
+// 2) it's Ready condition is set to true
+// 3) doesn't have NetworkUnavailable condition set to true
+func isNodeSchedulable(node *corev1.Node) bool {
+	nodeReady := isNodeConditionSetAsExpected(node, corev1.NodeReady, true, false)
+	networkReady := isNodeConditionUnset(node, corev1.NodeNetworkUnavailable) ||
+		isNodeConditionSetAsExpected(node, corev1.NodeNetworkUnavailable, false, true)
+	return !node.Spec.Unschedulable && nodeReady && networkReady
+}
+
+// Test whether a fake pod can be scheduled on "node", given its current taints.
+func isNodeUntainted(node *corev1.Node) bool {
```

Comment:
I don't like this amount of dependencies.

But in general, this is a bit of overkill. You can pretty much copy this one instead:
https://github.com/kubernetes/kubernetes/blob/807eb6b24b953ff3f59c8049eff7e0edbc279e18/pkg/scheduler/algorithm/predicates/predicates.go#L1506

and you won't even need those dependencies at all.

Reply:
Done.
I've also copied variables from `pkg/controller/nodelifecycle`, so dependency to the package could be removed.